ReactiveXComponent/WebSocket/WebSocketXCApiManager.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Threading;
+using Newtonsoft.Json;
+using ReactiveXComponent.Common;
+
+namespace ReactiveXComponent.WebSocket
+{
+    public class WebSocketXCApiManager
+    {
+        private readonly IWebSocketClient _webSocketClient;
+
+        private event EventHandler<WebSocketGetXcApiListResponse> XCApiListReceived;
+        private event EventHandler<WebSocketGetXcApiResponse> XCApiReceived;
+
+        private readonly IObservable<WebSocketGetXcApiListResponse> _xcApiListStream;
+        private readonly IObservable<WebSocketGetXcApiResponse> _xcApiStream;
+
+        public WebSocketXCApiManager(IWebSocketClient webSocketClient)
+        {
+            _webSocketClient = webSocketClient;
+
+            _xcApiListStream = Observable.FromEvent<EventHandler<WebSocketGetXcApiListResponse>, WebSocketGetXcApiListResponse>(
+                handler => (sender, e) => handler(e),
+                h => XCApiListReceived += h,
+                h => XCApiListReceived -= h);
+
+            _xcApiStream = Observable.FromEvent<EventHandler<WebSocketGetXcApiResponse>, WebSocketGetXcApiResponse>(
+                handler => (sender, e) => handler(e),
+                h => XCApiReceived += h,
+                h => XCApiReceived -= h);
+        }
+
+        public List<string> GetXCApiList(string requestId = null, TimeSpan ? timeout = null)
+        {
+            var delay = timeout ?? TimeSpan.FromSeconds(10);
+            List<string> result = null;
```

Comment:
it's a better practice to return an empty list in case we don't get a response in the timeout